### PR TITLE
Add pecl support.

### PR DESCRIPTION
**Pull request by jmjoy**
1. The pecl package need to call these steps:
    
    ```shell
    phpize
    ./configure
    make
    make install
    ```
     
    So add `config.m4` for `phpize`.

2. The `package.xml` is the pecl package definition file, the `<date>` and `<notes>` should be changed before release.

### CODE REVIEW

File: package.xml
Comment (jmjoy) [inline package.xml:18]: @heyanlong You should add you name in this location.

File: package.xml
Comment (heyanlong) [inline package.xml:38]: The block should include all source files

File: package.xml
Comment (jmjoy) [inline package.xml:105]: Is `<changelog>` necessary? I found that swoole was not written, and this was a little cumbersome and superfluous.

Comment (heyanlong) [reply package.xml:105]: This is not required, PECL will record change log.

**Comment by heyanlong** [comment]
PECL package should from GitHub Action Artifacts (tag only). After the CI I will upload it to PECL website and send release note to PECL developers list

File: package.xml
Comment (jmjoy) [reply package.xml:105]: I found swoole was not written, but it has:

Comment (heyanlong) [reply package.xml:105]: Maybe record from `notes` and `version` tag

Comment (jmjoy) [reply package.xml:105]: Delete it and try try the `pecl.php.net` will or will not has the `<changelog>`?

Comment (heyanlong) [reply package.xml:105]: Yes, Delete it

File: package.xml
Comment (wu-sheng) [reply package.xml]: According to ASF policy(ref to Java repo root pom.xml), the author should be Apache SkyWalking, the mail should be dev mail list.

Comment (wu-sheng) [reply package.xml]: I feel it is fine to add your names, but please make sure SkyWalking is listed on the top at least.

Comment (heyanlong) [reply package.xml]: @jmjoy SkyWalking can be lead, we're developer

Comment (heyanlong) [reply package.xml]: Like this
```
<lead>
<name>SkyWalking</name>
<email>dev@skywalking.apache.org</email>
</lead>
<lead>
<name>Yanlong He</name>
<user>yanlong</user> <!-- The user block is PECL account ID -->
<email>yanlong@php.net</email>
</lead>

```

Comment (jmjoy) [reply package.xml]: Alright, but I don't have pecl account.

Comment (jmjoy) [reply package.xml]: Does the PECL check the user field exists as a PECL account?

Comment (jmjoy) [reply package.xml]: Changed.

File: package.xml
Comment (wu-sheng) [inline package.xml:42]: Let's add a kind of anchor here? And write a pre-release script to update this automatically?
Then put this script in the pre-release steps of the release guide(Like Rust's?)

File: package.xml
Comment (wu-sheng) [inline package.xml:30]:
```suggestion
        <!-- SkyWalking committers could be listed here as core maintainers. -->
	<developer>
```

File: package.xml
Comment (jmjoy) [reply package.xml:43]: I think we can create a `package.tpl.xml` file and put `package.xml` to `.gitignore`, and use a script to generate the `package.xml`.

Because `<notes>` needs the changelog, so we can add a CI, trigger by the pre-release event, and auto fetch the changelog content to replace it.

@heyanlong What's your idea?

Comment (wu-sheng) [reply package.xml:43]: I think tpl.xml should be good too. We use `makefile` as the default compiling entrance anyway.

Comment (jmjoy) [reply package.xml:43]: `Makefile` is occupied by PECL and phpize (auto generate), so we can't use it...

We can add a [[bin]] in Cargo.toml, and then call `cargo run` to generate the `package.xml`.

Comment (wu-sheng) [reply package.xml:43]: Once we are clear about how to build in the docs, we should be good. Either makefile or cargo run is a good tooling.

Comment (jmjoy) [reply package.xml:43]: :ok:

Comment (jmjoy) [reply package.xml:43]: I also will generate the git tree file list for `<contents>`, now the field to be auto generated is `<date>`, `<version>`, `<notes>` and `<contents>`, alright?

Comment (wu-sheng) [reply package.xml:43]: Things you plan to change before release, we should be good. I just don't want you to have to re-run all release processes, because of forgetting these steps.

Comment (heyanlong) [reply package.xml:43]: tpl.xml is a good idea

Comment (jmjoy) [reply package.xml:43]: :ok_hand:

**Comment by jmjoy** [comment]
The pecl support is ready, the documents change will be added in next PRs.

**Comment by heyanlong** [comment]
No artifacts output?

**Comment by jmjoy** [comment]
What artifacts?

**Comment by heyanlong** [comment]
https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts

**Comment by jmjoy** [comment]
I just add tests, artifacts should be generate for tag event?

**Comment by heyanlong** [comment]
Yes, following tag event.

**Comment by jmjoy** [comment]
Add tag event in future?

**Comment by heyanlong** [comment]
Can also

**Comment by wu-sheng** [comment]
What do we discuss here? Do you try to build nightly build or build per commit?

**Comment by jmjoy** [comment]
> What do we discuss here? Do you try to build nightly build or build per commit?

I prefer to build artifacts from CI by tag event, no nightly build.

**Comment by wu-sheng** [comment]
The issue would be how should we call that. The RC binary should be built and signed by release manager manually, that is ASF release policy.

**Comment by jmjoy** [comment]
> The issue would be how should we call that. The RC binary should be built and signed by release manager manually, that is ASF release policy.

Yes, maybe it's not necessary to build artifacts by tag event, or only build the package, not the sign. The signed and later operations must be handle manually.

**Comment by wu-sheng** [comment]
> > The issue would be how should we call that. The RC binary should be built and signed by release manager manually, that is ASF release policy.
> 
> Yes, maybe it's not necessary to build artifacts by tag event, or only build the package, not the sign. The signed and later operations must be handle manually.

This is not very good practice. Maybe acceptable only? Because you should only sign a thing built manually rather than from a 3rd party. Such as one day, if GitHub mixed some things as CI is not a hard-protected env(or may be hacked), then we signed and released a totally unexpected binary.

**Comment by jmjoy** [comment]
> > > The issue would be how should we call that. The RC binary should be built and signed by release manager manually, that is ASF release policy.
> > 
> > 
> > Yes, maybe it's not necessary to build artifacts by tag event, or only build the package, not the sign. The signed and later operations must be handle manually.
> 
> This is not very good practice. Maybe acceptable only? Because you should only sign a thing built manually rather than from a 3rd party. Such as one day, if GitHub mixed some things as CI is not a hard-protected env(or may be hacked), then we signed and released a totally unexpected binary.

Alright, we will not build artifacts from CI. @heyanlong

File: scripts/README.md
Comment (wu-sheng) [inline scripts/README.md:1]: What is this file about?

**Comment by wu-sheng** [comment]
I can see you update the scripts, but I can't find the release doc, compiling doc, and pecl install doc udpate.

File: scripts/README.md
Comment (jmjoy) [reply scripts/README.md:1]: It's just a file that occupies a place and has no purpose, can be deleted.

**Comment by jmjoy** [comment]
> I can see you update the scripts, but I can't find the release doc, compiling doc, and pecl install doc udpate.

I want to update the doc next PRs.

File: scripts/README.md
Comment (wu-sheng) [reply scripts/README.md:1]: Is this the file which would be located in the source tar or binary tar?

Comment (jmjoy) [reply scripts/README.md:1]: Yes, it would.

Comment (wu-sheng) [reply scripts/README.md:1]: OK, if you want to keep this file, we should add a standard readme, which is similar with project's readme in the GitHub.

Comment (jmjoy) [reply scripts/README.md]: I have deleted it.

**Review by wu-sheng — APPROVED** [review]

**Comment by wu-sheng** [comment]
BTW, you should try to set up the required CI tasks to block merging before CI passed.

**Comment by jmjoy** [comment]
> BTW, you should try to set up the required CI tasks to block merging before CI passed.

OK, I have some doubts about this. I'll study it later.

**Pull request merged**
